src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a.toString() > b.toString()) {
+        return 1;
+      } else if (a.toString() < b.toString()) {
+        return -1;
+      } else {
+        return 0;
+      }
+    },
+  ) {
+    let tempValue;
+
+    for (let i = 0; i < this.length; i++) {
+      for (let k = 0; k < this.length - 1; k++) {
+        if (compareFunction(this[k], this[k + 1]) > 0) {
+          tempValue = this[k];
+          this[k] = this[k + 1];
+          this[k + 1] = tempValue;
+        }
+      }
+    }
```

Comment:
The bubble sort implementation can be optimized by reducing the number of iterations in the inner loop as the largest elements 'bubble up' to the end of the array with each outer loop iteration. You can subtract 'i' from the inner loop's condition to avoid unnecessary comparisons.